encord/metadata_schema.py

```diff
@@ -311,14 +339,41 @@ def delete_key(self, k: str) -> None:
 
         MetadataSchemaError: If the key `k` is already deleted or not present in the schema
         """
+        if k not in self._schema:
+            raise MetadataSchemaError(f"{k} is not defined")
+        _assert_valid_metadata_key(k)
+        deleted_ty = self._schema[k].root
+        if isinstance(deleted_ty, _ClientMetadataSchemaTypeTombstone):
+            raise MetadataSchemaError(f"{k} is already deleted")
+        self._schema[k] = _ClientMetadataSchemaOption(
+            root=_ClientMetadataSchemaTypeTombstone(
+                deleted_ty=None if hard else _ClientMetadataSchemaTypeTombstoneDeletedTy(root=deleted_ty)
+            )
+        )
+        self._dirty = True
+
+    def restore_key(self, k: str) -> None:
+        """
+        Restore a deleted metadata key to its original value.
+
+        **Parameters:**
+
+        k : str: The key for which the metadata type is to be restored.
+
+        **Raises:**
+
+        MetadataSchemaError: If the key `k` is not already deleted or not present in the schema
+        """
         if k in self._schema:
             v = self._schema[k]
-            if isinstance(v.root, _ClientMetadataSchemaTypeTombstone):
-                raise MetadataSchemaError(f"{k} is already deleted")
+            tombstone_ty = v.root
+            if not isinstance(tombstone_ty, _ClientMetadataSchemaTypeTombstone):
+                raise MetadataSchemaError(f"{k} is not currently deleted")
         else:
```

Comment:
Insanely styley, can dismiss:
Check k not in self._schema and do early raise rather than if,else